src/hooks/useUnreadMessages.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState, createContext, useContext, useEffect } from 'react';
+import { onNotificationReceived } from '../common/Notifications';
+import { uniq } from 'lodash';
+import { useAsyncStorage } from './useAsyncStorage';
+
+type UnreadMessageContextProps = {
+  unreadMessagesUserIds?: string[];
+  markMessageRead?: (userId: string) => void;
+};
+
+const UnreadMessagesContext = createContext<UnreadMessageContextProps>({});
+
+export const UnreadMessagesContextProvider = ({
+  children,
+}: {
+  children?: React.ReactNode;
+}) => {
+  const [initialUnreadMessages, setStoredUnreadMessages] = useAsyncStorage(
+    'UnreadMessageUserIds',
+  );
+
+  const [unreadMessagesUserIds, setUnreadMessages] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (initialUnreadMessages.isFetchedAfterMount) {
+      setUnreadMessages(
+        initialUnreadMessages.data
+          ? initialUnreadMessages.data?.split('#')
```

Comment:
Why split by `#`? Why not just stringify the JSON?

Reply:
That was a pretty obvious miss on my part, I will update.

Reply:
I wonder if it would make sense to move the serialization into `useAsyncStorage`.

Reply:
I attempted to move this into useAsyncStorage for this PR but hit a bit of a wall on deserialization and type casting, decided to revisit this later.